bindings/src/main/java/io/pravega/storage/azure/AzureChunkStorage.java

```diff
@@ -0,0 +1,210 @@
+        this.closed = new AtomicBoolean(false);
+        this.shouldClose = shouldClose;
+        this.supportsAppend = supportsAppend;
+
+    }
+
+    @Override
+    public boolean supportsTruncation() {
+        return false;
+    }
+
+    @Override
+    public boolean supportsAppend() {
+        return supportsAppend;
+    }
+
+    @Override
+    public boolean supportsConcat() {
+        return false;
+    }
+
+    @Override
+    protected ChunkInfo doGetInfo(String chunkName) throws ChunkStorageException {
+        try {
+            val blobProperties = client.getBlobProperties(getObjectPath(chunkName));
+            return ChunkInfo.builder()
+                    .name(chunkName)
+                    .length(blobProperties.getBlobSize())
+                    .build();
+        } catch (Exception e) {
+            throw convertException(chunkName, "doGetInfo", e);
+        }
+    }
+
+    @Override
+    protected ChunkHandle doCreate(String chunkName) throws ChunkStorageException {
+        try {
+            val blobItem = this.client.create(getObjectPath(chunkName));
+            return ChunkHandle.writeHandle(chunkName);
+        } catch (Exception e) {
+            throw convertException(chunkName, "doCreate", e);
+        }
+    }
+
+    @Override
+    protected boolean checkExists(String chunkName) throws ChunkStorageException {
+        return client.exists(getObjectPath(chunkName));
+    }
+
+    @Override
+    protected void doDelete(ChunkHandle handle) throws ChunkStorageException {
+        try {
+            client.delete(getObjectPath(handle.getChunkName()));
+        } catch (Exception e) {
+            throw convertException(handle.getChunkName(), "doDelete", e);
+        }
+    }
+
+    @Override
+    protected ChunkHandle doOpenRead(String chunkName) throws ChunkStorageException {
+        if (!checkExists(chunkName)) {
+            throw new ChunkNotFoundException(chunkName, "doOpenRead");
+        }
+        return ChunkHandle.readHandle(chunkName);
+    }
+
+    @Override
+    protected ChunkHandle doOpenWrite(String chunkName) throws ChunkStorageException {
+        if (!checkExists(chunkName)) {
+            throw new ChunkNotFoundException(chunkName, "doOpenWrite");
+        }
+        return new ChunkHandle(chunkName, false);
+    }
+
+    @Override
+    protected int doRead(ChunkHandle handle, long fromOffset, int length, byte[] buffer, int bufferOffset) throws ChunkStorageException {
+        try  {
+            try (val inputStream = client.getInputStream(getObjectPath(handle.getChunkName()), fromOffset, length)) {
+                val retValue =  StreamHelpers.readAll(inputStream, buffer, bufferOffset, length);
+                if (retValue == 0) {
+                    val blobProperties = client.getBlobProperties(getObjectPath(handle.getChunkName()));
+                    if (blobProperties.getBlobSize() <= fromOffset || blobProperties.getBlobSize() <= fromOffset + length) {
```

Comment:
Does this check that having `retValue == 0` is only valid in case of the chunk being empty, right?

Reply:
Yes. It suggests that the bytes read is zero.